ZohoCRMCharts/Classes/ZCRMFunnel.swift

```diff
@@ -0,0 +1,1148 @@
+		} else if self.type == .classic {
+			self.addClassicFunnelConstraints()
+		} else if self.type == .segment {
+			self.addSegmentFunnelConstraints()
+		} else {
+			self.addStandardFunnelConstraints()
+		}
+	}
+	
+	func updateChanges() {
+		
+		switch self.type {
+		case .path:
+			self.updatePathUIOptions()
+		case .compact:
+			self.updateCompactFunnelUIOptions()
+		case .classic:
+			self.updateClassicFunnelUIOptions()
+		case .segment:
+			self.updateSegmentUIOptions()
+		case .standard:
+			self.updateStandardUIOptions()
+		}
+	}
+	
+	private func renderView() {
+		
+		if self.type == .path {
+			self.renderPathFunnel()
+		} else if self.type == .compact {
+			self.renderCompactFunnel()
+		} else if self.type == .classic {
+			self.renderClassicFunnel()
+		} else if self.type == .segment {
+			self.renderSegmentFunnel()
+		} else  {
+			self.renderStandardFunnel()
+		}
+	}
+	
+	private func renderTitleView() {
+		self.setTitleView()
+		self.updateTitleUIOptions()
+	}
+	
+	func updateTitleUIOptions() {
+		
+		self.titleView.font = self.renderOptions.titleFont
+		self.titleView.textColor = self.renderOptions.titleFontColor
+	}
+	
+	func updateConverionRateViewUIOptions() {
+		
+		if self.type == .standard {
+			self.standardConversionRateView.font = self.renderOptions.conversionRateFont
+			self.standardConversionRateView.fontColor = self.renderOptions.conversionRateFontColor
+			self.standardConversionRateView.setUIOptions()
+		} else {
+			self.conversionRateView.font = self.renderOptions.conversionRateFont
+			self.conversionRateView.textColor = self.renderOptions.conversionRateFontColor
+		}
+	}
+	
+	private func setTitleView() {
+		
+		self.titleView.translatesAutoresizingMaskIntoConstraints = false
+		self.titleView.text = self.title
+		self.addSubview(self.titleView)
+		var constraints: [NSLayoutConstraint] = []
+		constraints += NSLayoutConstraint.constraints(withVisualFormat: "H:|-15-[title]", options: [], metrics: nil, views: ["title": self.titleView])
+		constraints += NSLayoutConstraint.constraints(withVisualFormat: "V:|-15-[title(>=25)]", options: [], metrics: nil, views: ["title": self.titleView])
+		NSLayoutConstraint.activate(constraints)
+	}
+	
+	private func addConverionRateView() {
+		
+		self.updateConverionRateViewUIOptions()
+		self.conversionRateView.translatesAutoresizingMaskIntoConstraints = false
+		self.conversionRateView.textAlignment = .center
+		self.conversionRateView.numberOfLines = 1
+		self.addSubview(self.conversionRateView)
+	}
+	
+	private func renderPathFunnel() {
+		
+		self.scrollView.translatesAutoresizingMaskIntoConstraints = false
+		self.addSubview(self.scrollView)
+		
+		self.stackView1.translatesAutoresizingMaskIntoConstraints = false
+		self.stackView1.axis = .horizontal
+		self.stackView1.distribution = .fillEqually
+		self.scrollView.addSubview(self.stackView1)
+		
+		self.stackView2.translatesAutoresizingMaskIntoConstraints = false
+		self.stackView2.axis = .horizontal
```

Comment:
use different naming instead of stackView1 and stackView2.

Reply:
The stack views have been used for different aspects all over the class. To be generic it had named like that.